packages/app/src/signIn/SignInScreen.tsx

```diff
@@ -95,11 +104,29 @@ export default function SignInScreen() {
   }, [authenticationClient, overrideEmailAddress]);
 
   const userRecord = useCurrentUserRecord(authenticationClient);
+
+  React.useEffect(() => {
```

Comment:
I think this effect can be merged into the the one on line 123, which might make this state machine a bit clearer. Boy, effect hooks do start to fall apart when you get a bunch of them together like this.